### PR TITLE
PRD-978 fix card auth rejection handling

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1140 fix device upgrade redirect bugs & card auth](https://app.clickup.com/t/37308523/CSI-1140)

File: src/flows/cardAuth/index.ts [hunk 1]
```diff
@@ -155,7 +155,7 @@ export class CardAuthenticator extends CyFlow {
     });
 
     if (cardData.commandType === 70 && cardData.data.startsWith('00')) {
-      this.emit('cardError');
+      this.emit('acceptedRequest', false);
       throw new ExitFlowError();
     }
```
Comment (irshadCypherock) [inline src/flows/cardAuth/index.ts:160]: Need `cardError` handling as well.

Comment (HarishRocks) [reply src/flows/cardAuth/index.ts:160]: but how?
```
const acceptedRequest = await connection.receiveData([70], 30000);
if (acceptedRequest.data === '00') {
  this.emit('acceptedRequest', false);
  throw new ExitFlowError();
} else {
  this.emit('acceptedRequest', true);
}

const receivedHash = await connection.receiveData([13, 70], 90000);

if (receivedHash.commandType === 70 && receivedHash.data.startsWith('00')) {
  this.emit('cardError');
  throw new ExitFlowError();
} else if (receivedHash.commandType !== 13) {
  throw new Error('Invalid command received');
}
```

In the legacy code, we are receive command 70 twice. But in the new function, it is only received first

Comment (irshadjsr21) [reply src/flows/cardAuth/index.ts:160]: Need to discuss this with device team

Comment (HarishRocks) [reply src/flows/cardAuth/index.ts:160]: okayy

Comment (ujjwal-cyph) [reply src/flows/cardAuth/index.ts:160]: There is one byte payload along with the Command ID 70. Can you use that? Both instances will have different payload byte.

Comment (ujjwal-cyph) [reply src/flows/cardAuth/index.ts:160]: IIRC, the payload byte with value `00` is supposed to indicate critical failure. So the process will be exited and device will go to default screen.

Comment (HarishRocks) [reply src/flows/cardAuth/index.ts:160]: After speaking to @ujjwal-cyph ,
he mentioned that device would sending error codes and messages to desktop, the implementation is in progress. So we can pick it up after that @irshadCypherock. Until then the desktop wouldn't be able to handle failure errors.

Comment (irshadCypherock) [reply src/flows/cardAuth/index.ts:160]: Okay, merging this for now. @HarishRocks Please create a ticket for handling card and other errors.

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**